FWCore/Framework/src/GenericHandle.cc

```diff
@@ -73,4 +73,18 @@ namespace edm {
     return false;
   }
 
+  ///Specialize the Event's getByToken method to work with a Handle<GenericObject>
+  template <>
+  bool Event::getByToken<GenericObject>(EDGetToken token, Handle<GenericObject>& result) const {
+    result.clear();
```

Comment:
As @fwyzard mentioned in mattermost, the implementations of the `getByLabel()` specializations have diverged from the generic `getByLabel()`, whereas the added `getByToken()` specialization follows the structure of the generic `getByToken()`. @Dr15Jones would you agree it would be good to update the `getByLabel()` specializations (in a separate PR unless @fwyzard would be willing to do that here as well)

Reply:
If the differences are not actually functional (I can't really tell with a cursory look) than an update to make it consistent would make sense.

Reply:
The only possibly functional difference I see is that the generic `getByLabel()` has this `result.clear()` call, while the specializations above do not have it.